### PR TITLE
Offer Reset: create a filter bar for the new Plans page

### PR DESCRIPTION
**Pull request by rcanepa**
#### Changes proposed in this Pull Request

* Create a filter bar that includes a product type filter (Security, Performance, All) and a duration (monthly, annually) filter.

#### Note

* This doesn't include the logic to make the bar sticky. It might be better to do that in another PR once we have added the Cart bar (the one in the top of this one) to the page because both bars should be sticky.

#### Testing instructions

* Run this PR.
* Activate the `offerResetFlow` AB test.
* Visit `http://calypso.localhost:3000/plans/:site`.
* Check both selectors work and they look as per design.

Fixes 1169247016322522-as-1187323761633614

#### Demo
image
image
Kapture 2020-08-05 at 15 07 27

### CODE REVIEW

File: client/my-sites/plans-v2/plans-filter-bar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+import React, { useState } from 'react';
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import SegmentedControl from 'components/segmented-control';
+import SelectDropdown from 'components/select-dropdown';
+import { TERM_MONTHLY, TERM_ANNUALLY } from 'lib/plans/constants';
+import type { Duration, ProductType } from '../types';
```
Comment (ChaosExAnima) [inline client/my-sites/plans-v2/plans-filter-bar/index.tsx:14]: It's my personal taste- I like having a separate "Type dependencies" area to clearly distinguish types vs actual imports.

File: client/my-sites/plans-v2/plans-filter-bar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+		label: translate( 'All' ),
+	},
+};
+
+interface Props {
+	duration: Duration;
+	productType: ProductType;
+	onDurationChange: Function;
+	onProductTypeChange: Function;
+}
```
Comment (ChaosExAnima) [inline client/my-sites/plans-v2/plans-filter-bar/index.tsx:42]: Let's keep new types in the `types.ts`?

File: client/my-sites/plans-v2/plans-filter-bar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+	},
+	[ PERFORMANCE ]: {
+		id: PERFORMANCE,
+		label: translate( 'Performance' ),
+	},
+	[ ALL ]: {
+		id: ALL,
+		label: translate( 'All' ),
+	},
+};
```
Comment (ChaosExAnima) [inline client/my-sites/plans-v2/plans-filter-bar/index.tsx:35]: Maybe should be in the `constants` file?

File: client/my-sites/plans-v2/plans-filter-bar/style.scss [hunk 4]
```diff
@@ -0,0 +1,75 @@
+.plans-filter-bar {
+	$height: 36px;
+	// SegmentedControl uses a border-radius of 4px while our design specifies 3px
```
Comment (monsieur-z) [inline client/my-sites/plans-v2/plans-filter-bar/style.scss:3]: I'd say it's okay to use the border radius of the current component. It makes the styles leaner and the components more consistent across the app.

**Comment by monsieur-z** [comment]
Some minor comments about the UI:
- Some padding between the 2 controls on mobile would be nice (viewport in screenshot is 375px)
Screen Shot 2020-08-05 at 4 04 20 PM
- In what seems to be the `:active` or pressed down state, the color of the dropdown top text is too dark
Screen Shot 2020-08-05 at 4 03 44 PM

File: client/my-sites/plans-v2/plans-filter-bar/index.tsx (hunk 3 above)
Comment (rcanepa) [reply client/my-sites/plans-v2/plans-filter-bar/index.tsx:35]: Good idea.

File: client/my-sites/plans-v2/plans-filter-bar/index.tsx (hunk 1 above)
Comment (rcanepa) [reply client/my-sites/plans-v2/plans-filter-bar/index.tsx:14]: I like that too.

File: client/my-sites/plans-v2/constants.ts [hunk 5]
```diff
@@ -0,0 +1,3 @@
+export const ALL = 'all';
+export const PERFORMANCE = 'performance';
+export const SECURITY = 'security';
```
[the remark this answers is not in the document]
Comment (rcanepa) [reply client/my-sites/plans-v2/constants.ts:3]: I don't understand what you mean. Are you referring to the new constants being added in #44705?

File: client/my-sites/plans-v2/plans-filter-bar/index.tsx (hunk 2 above)
Comment (rcanepa) [reply client/my-sites/plans-v2/plans-filter-bar/index.tsx:42]: But this is the definition of the interface for the component props. Am I misreading your comment?

File: client/my-sites/plans-v2/plans-filter-bar/style.scss (hunk 4 above)
Comment (rcanepa) [reply client/my-sites/plans-v2/plans-filter-bar/style.scss:3]: Good advice. It's hard to find the right balance between being consistent with what we already have and following the designs.

**Comment by rcanepa** [comment]
I have push several updates. In terms of the UI, I made sure that in small screens both filters never get touch each other and I fixed wrong colors on the product type selector.

Here is how the product type selector should look like now:

image

How the bar looks like in different screen sizes: 

Kapture 2020-08-06 at 13 06 16

Another option would be to stack them on very small screen.

File: client/my-sites/plans-v2/constants.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (ChaosExAnima) [reply client/my-sites/plans-v2/constants.ts:8]: Yeah, I meant eventually! You're good here.

**Pull request merged**